### DISCUSSION TITLE
프로젝트 성능 개선 회의

### DISCUSSION THREAD

**Discussion by her0807** [woowacourse-teams/2022-ternoko#351] [Announcements]
성능 개선하고 싶은 부분에 대해 자유롭게 의견 나눈다.

**Comment by dongho108** [comment]
인터뷰 상태를 디자인 패턴으로 풀고 싶슴다!

**Comment by dongho108** [comment]
면담가능시간에 인덱싱 걸어보시죠

**Reply by soominsohn** [reply]
또 어디에 인덱싱 걸을 수 있을지 공부하고 찾아보고 싶습니다!

**Comment by lokba** [comment]
프론트는 추석을 이용해서 성능 개선 포인트 전반적으로 정리하겠습니다.

**Comment by soominsohn** [comment]
수달이 말했던 것처럼 
서비스에 있는 것들 도메인으로 옮길수 있는 거 더 있나 찾아보고
예외에 대한 테스트 케이스 보강하고 싶어요~

**Comment by soominsohn** [comment]
DB Replication 도입

**Comment by soominsohn** [comment]
테스트 관련

메소드나 api 요청 별로 시간 어느정도 걸리는지 측정해보고 싶습니다. 

또 부하테스트도 해봐서 어느정도 요청 들어왔을 때 뻗는지 파악하고, 서버 장애 났을 때 어떻게 대응할지 전략 수립도 해보고 싶습니다.

**Comment by dongho108** [comment]
시간관련 테스트 리팩토링 해야할 것 같아용

**Comment by dongho108** [comment]
현재 사용중인 금지기술 설득 or 교체

- WebClient
   - restTemplate 과 비교해서 쓴 이유
   - 아니면 교체하기
- EventListener
   - 이벤트리스너의 원래용도
   - 지금 필요한이유
   - 아니면 교체하기

**Reply by her0807** [reply]
수달

**Reply by Juhyung990122** [reply]
이벤트가 금지기술이라 걷는게 맞긴 할텐데 JPA에서도 도메인이벤트 기능 어노테이션을 제공하는 것 같네요~! 
대충봐서 정확한건진 모르겠지만 알아두면 좋을 듯 합니다!

https://velog.io/@max9106/JPA-%EB%8F%84%EB%A9%94%EC%9D%B8-%EC%9D%B4%EB%B2%A4%ED%8A%B8
요약 : save() 할때 이벤트 발행됨.
고민 : JPA의 더티체킹을 활용하는 로직에서는 이벤트 발행이 안될듯?(뇌피셜입니다.)

**Comment by dongho108** [comment]
N + 1 발생하는 곳 찾고 해결

**Reply by her0807** [reply]
수달

**Comment by her0807** [comment]
clock 으로 시간 테스트하는 로직 전반적으로 리팩터링

**Comment by dongho108** [comment]
백에서 보내주는 예외메세지를 프론트에서 볼 수 있도록 하기

**Comment by soominsohn** [comment]
젠킨스 master-slave 도입하고 싶습니다

**Reply by her0807** [reply]
젠킨스 하나가 터져도 장애에 지장 없게

**Reply by Juhyung990122** [reply]
젠킨스 하나씩 접속해서 관리하지 않고  master 하나에서 제어할 수 있어서 관리측면에서도 좋을듯하네요~

**Comment by her0807** [comment]
도메인 테스트 코드 보강
